crates/issue/src/lib.rs

```diff
@@ -123,7 +122,31 @@ pub mod pallet {
     }
 
     #[pallet::hooks]
-    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<(), &'static str> {
+            ensure!(Self::storage_version() == Version::V0, "Wrong storage version.");
+            let issue_count_before = crate::IssueRequests::<T>::iter().count() as u32;
+            log::info!("issue count before: {:?}", issue_count_before);
+            crate::types::v4::IssueCountBefore::<T>::put(issue_count_before);
```

Comment:
Yea I know, I spent awhile thinking what else to check but it would be hard to do specific data validation inside the pallet - we could move to runtime maybe but not sure it's worth it.